src/interpreter/bytecode-array-builder.cc

```diff
@@ -1412,7 +1414,9 @@ BytecodeArrayBuilder& BytecodeArrayBuilder::RecordReplayInstrumentationGenerator
   if (emit_record_replay_opcodes_ && (recordreplay::IsReplaying() || gRecordReplayAssertValues)) {
     int bytecode_offset = bytecode_array_writer_.size();
     int index = RegisterInstrumentationSite(kind, kNoSourcePosition, bytecode_offset);
-    OutputRecordReplayInstrumentationGenerator(index, generator_object);
+    if (index > 0) {
```

Comment:
How is index ever 0 (or negative)?  Looking at RegisterInstrumentationSite, I see `return index + BytecodeSiteOffset`, and `BytecodeSiteOffset` is 65536, and `index` is the size of a vector.

Reply:
@klochek 
There is a new early-return [here](https://github.com/replayio/chromium-v8/pull/189/files#diff-f2bb4c913b64187a385ad088690871139a6f37c8c6b1a0ce02decb6f808bc669R1351)

Reply:
Thank you, totally missed that :P  Now I don't understand why the gInstrumentationSites global is _not_ a part of the bytecode generator, but that's not your problem.